src/OpenIddict.Server.Quartz/OpenIddictServerQuartzJob.cs

```diff
@@ -0,0 +1,154 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/openiddict/openiddict-core for more information concerning
+ * the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OpenIddict.Abstractions;
+using Quartz;
+using static OpenIddict.Abstractions.OpenIddictExceptions;
+using SR = OpenIddict.Abstractions.OpenIddictResources;
+
+namespace OpenIddict.Server.Quartz
+{
+    /// <summary>
+    /// Represents a Quartz.NET job performing scheduled tasks for the OpenIddict server feature.
+    /// </summary>
+    [DisallowConcurrentExecution]
+    public class OpenIddictServerQuartzJob : IJob
+    {
+        private readonly ILogger _logger;
+        private readonly IOptionsMonitor<OpenIddictServerQuartzOptions> _options;
+        private readonly IServiceProvider _provider;
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="OpenIddictServerQuartzJob"/> class.
+        /// </summary>
+        public OpenIddictServerQuartzJob() => throw new InvalidOperationException(SR.GetResourceString(SR.ID1081));
```

Comment:
Quartz options for job factory has `AllowDefaultConstructor` which defaults to false, so Quartz shouldn't call this by default. I think could also be private?

Reply:
Ah you now when I track this back it seems that you want to catch invalid configuration, neat.

Reply:
Absolutely, the parameterless constructor is just here to inform the user that enabling DI is mandatory for the OpenIddict server job to work correctly. Once DI is enabled, the other constructor will always be preferred (whether you're using the MS DI or any other complying DI like Autofac).